### PR TITLE
 Add PKGBUILD for Arch Linux 

### PR DESCRIPTION
**Pull request by Maryse47**
This template will allow Arch Linux users install this project very easily.

To make things work properly fixes for https://github.com/mjec/fido2-hmac-secret/issues/8 and https://github.com/mjec/fido2-hmac-secret/issues/9 are needed.

### CODE REVIEW

File: contrib/archlinux/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,30 @@
+pkgname=fido2-hmac-secret-git
+_pkgname=fido2-hmac-secret
+pkgver=0.3.0.r12.ge474de5
+pkgrel=1
+pkgdesc="A simple way to generate password-proteceted secrets from a FIDO2 authenticator with the hmac-secret extension"
+arch=('x86_64')
+url="https://github.com/mjec/fido2-hmac-secret"
+license=('GPL3')
+depends=('bash' 'libfido2' 'libcbor' 'libsodium' 'libcap')
+makedepends=('clang' 'git')
```
Comment (mjec) [inline contrib/archlinux/PKGBUILD:10]: Don't need clang if using `CC=gcc`. It will be a `checkdepends`, but we don't have a `check()` yet (and I don't think we want one in the PKGBUILD right now)

File: contrib/archlinux/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,30 @@
+pkgname=fido2-hmac-secret-git
+_pkgname=fido2-hmac-secret
+pkgver=0.3.0.r12.ge474de5
+pkgrel=1
+pkgdesc="A simple way to generate password-proteceted secrets from a FIDO2 authenticator with the hmac-secret extension"
+arch=('x86_64')
+url="https://github.com/mjec/fido2-hmac-secret"
+license=('GPL3')
+depends=('bash' 'libfido2' 'libcbor' 'libsodium' 'libcap')
```
Comment (mjec) [inline contrib/archlinux/PKGBUILD:9]: I'd add optdepends on `bash-completion` and `mkinitcpio` (though why you'd be running arch without mkinitcpio I'm not sure)

File: contrib/archlinux/PKGBUILD (hunk 1 above)
Comment (mjec) [reply contrib/archlinux/PKGBUILD:10]: Can we make `libcap` a makedepends? Honestly it's going to be installed on most systems so I'm not super worried about it, but I'm also unclear on whether makedepends stick around for the `.install` step or not.

File: contrib/archlinux/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+pkgver() {
+  cd "$_pkgname"
+  git describe --long --tags | sed 's/^v//;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+build() {
+  cd "$_pkgname"
+  export CC="gcc"
+  make release bash-completion initcpio
```
Comment (mjec) [inline contrib/archlinux/PKGBUILD:24]: I'd do something like:

```sh
targets="release"

if pacman -Qq mkinitcpio >/dev/null 2>&1; then
  targets="$targets initcpio"
fi

if pacman -Qq bash-completion >/dev/null 2>&1; then
  targets="$targets bash-completion"
fi

make $targets
```

There's also an argument for changing the target name from `initcpio` to `mkinitcpio` here, and then we could loop over optdepends to append relevant targets... WDYT?

**Comment by Maryse47** [comment]
I think this is now complete.

File: contrib/archlinux/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+build() {
+  cd "$_pkgname"
+  make release bash-completion initcpio
+}
+
+package() {
+  cd "$_pkgname"
+  # Adding file capabilties doesn't work during build, it will be done later through fido2-hmac-secret.install
```
Comment (mjec) [inline contrib/archlinux/PKGBUILD:29]: No need to change this, but I'm curious as to why this doesn't work -- is it because `package` doesn't run as root?

**Review by mjec — COMMENTED** [review]
Thanks for putting this together! Getting this into AUR was a goal of mine, but I wasn't going to start on it until after tests (#6) and the other issues that are now closed.

A few suggestions, but generally looks good. I'm open to pushback on any of my comments, after which I'll merge.

If you want to add to AUR after this is merged please go ahead, otherwise I will do so. I'll also be tagging 0.4.0 (not waiting for #6) once this is merged.

File: contrib/archlinux/PKGBUILD (hunk 4 above)
Comment (Maryse47) [reply contrib/archlinux/PKGBUILD:29]: Yes, package (and whole build process doesn't run as root). Another thing is that AFAIK file capabilities aren't preserved during pack -> unpack process. Adding file capabilities during post-install is how official Arch packages handle this.

File: contrib/archlinux/PKGBUILD (hunk 1 above)
Comment (Maryse47) [reply contrib/archlinux/PKGBUILD:10]: The `CC=gcc` was leftover from testing and I didn't intent to put it there (it's removed now).

`makedepends` doesn't stick for  the `.install` as the latter is run only after package is installed which is separate process from building it. `libcap` is already satisfied in dependency chain so I could drop it completely however I wanted to make it visible here.

File: contrib/archlinux/PKGBUILD (hunk 2 above)
Comment (Maryse47) [reply contrib/archlinux/PKGBUILD:9]: Ok. Arch supports `dracut` which can be `mkinitcpio` replacement so I think running Arch without the latter is possible.

File: contrib/archlinux/PKGBUILD (hunk 3 above)
Comment (Maryse47) [reply contrib/archlinux/PKGBUILD:24]: I'm not a fan of that because there is no guarantee that environment where package is being build is the same as those where it will be installed i.e. building in clean chroot (which is mandatory if this will be ever put in official repos).

Moreover it's possible that user will install `bash-completion` after they build this one and in that case they would have to rebuild this package again to get functionality.

To sum up I think that build process should be deterministic.

Comment (mjec) [reply contrib/archlinux/PKGBUILD:24]: Alright I buy that

File: contrib/archlinux/PKGBUILD (hunk 2 above)
Comment (Maryse47) [reply contrib/archlinux/PKGBUILD:9]: optdepends added

**Comment by Maryse47** [comment]
For AUR you may also prepare non `-git` version basing on this one with a few tweaks. You may also consider signing tags.
